File: backend/app/models/puesto.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import relationship

from app.extensions import db


class Puesto(db.Model):
    __tablename__ = "puestos"

    id_puesto = db.Column(Integer, primary_key=True)
    nombre_puesto = db.Column(String(100), nullable=False)
    direccion = db.Column(String(200), nullable=False)
    estado = db.Column(String(20), nullable=False)

    dispositivos = relationship("Dispositivo", back_populates="puesto", lazy="select")
    turnos = relationship("Turno", back_populates="puesto", lazy="select")

    def __repr__(self) -> str:
        return f"<Puesto {self.nombre_puesto}>"

File: mobile/lib/features/positions/models/puesto.dart
class Puesto {
  const Puesto({
    this.idPuesto,
    required this.nombrePuesto,
    required this.direccion,
    required this.estado,
  });

  final int? idPuesto;
  final String nombrePuesto;
  final String direccion;
  final String estado;

  factory Puesto.fromJson(Map<String, dynamic> json) {
    return Puesto(
      idPuesto: json['id_puesto'] as int?,
      nombrePuesto: json['nombre_puesto'] as String,
      direccion: json['direccion'] as String,
      estado: json['estado'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'nombre_puesto': nombrePuesto,
        'direccion': direccion,
        'estado': estado,
      };

  Puesto copyWith({
    int? idPuesto,
    String? nombrePuesto,
    String? direccion,
    String? estado,
  }) {
    return Puesto(
      idPuesto: idPuesto ?? this.idPuesto,
      nombrePuesto: nombrePuesto ?? this.nombrePuesto,
      direccion: direccion ?? this.direccion,
      estado: estado ?? this.estado,
    );
  }
}
